src/ring_algs.rs

```diff
@@ -1,564 +1,15 @@
-// Copyright 2015 Brian Smith.
-//
-// Permission to use, copy, modify, and/or distribute this software for any
```

Comment:
Splitting the rename from `ring_algs.rs` to `ringlike_algs.rs` into a separate commit will make this a lot easier to review.